tests/fpsemi-examples.cpp

```diff
@@ -99,6 +99,114 @@ namespace libsemigroups {
         }
       }
     }
+    void add_Iwahori_full_transformation_monoid_relations(
+        std::vector<relation_type>& result,
+        size_t                      n,
+        size_t                      m) {
+      // This function adds the full transformation monoid relations due to
+      // Iwahori, from Section 9.3, p161-162, (Ganyushkin + Mazorchuk),
+      // expressed in terms of the generating set {pi_2, ..., pi_n,
+      // epsilon_{12}} using the notation of that chapter.
+      // https://link.springer.com/book/10.1007/978-1-84800-281-4
+
+      // The argument m specifies the letter value the idempotent e12
+      // corresponds to. When adding these relations for the full transformation
+      // monoid presentation, we want m = n - 1. For the partial transformation
+      // monoid presentaiton, we want m = n.
+
+      // It is useful to have this as a separate function, to avoid computing
+      // duplicate presentations for the underlying symmetric group, when
+      // combining relations from the symmetric inverse and full transformation
+      // monoids.
+
+      word_type              e12 = {m};
+      std::vector<word_type> pi;
+      for (size_t i = 0; i <= n - 2; ++i) {
+        pi.push_back({i});
+      }
+
+      // The following expresses the epsilon idempotents in terms of the
+      // generating set
+      auto eps = [&e12, &pi](size_t i, size_t j) -> word_type {
+        LIBSEMIGROUPS_ASSERT(i != j);
+        if (i == 1 and j == 2) {
```

Comment:
```suggestion
        if (i == 1 && j == 2) {
```

Reply:
Although the keywords `and`, `or`, and `not` exist in C++, we use `&&`, `||` and `!` instead.

Reply:
Done